### PR TITLE
Book list

### PR DESCRIPTION
**Pull request by coonik**
tao lai

### CODE REVIEW

File: Assignment/Book-Management/src/app/core/services/book.service.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+// export interface BookResponseData {
+//   idToken:	string,
+//   email: string,
+//   refreshToken: string,
+//   expiresIn: string,
+//   localId: string,
+//   register?: boolean
+// }
+
```
Comment (tranmanhlinh) [inline Assignment/Book-Management/src/app/core/services/book.service.ts:15]: Remove comment lines.

File: Assignment/Book-Management/src/app/core/services/book.service.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+@Injectable({ providedIn: 'root'})
+export class BookService {
+  constructor(private http: HttpClient, private route: Router) {}
+  getObs: Observable<any>;
+  totalPages: number;
+  totalItems: number;
+
+  getAllBook(pageIndex: number, pageSize: number) {
+
+    this.http.get<any>('https://nga-book-api.herokuapp.com/api/books?pageNumber='+pageIndex+'&pageSize='+pageSize, {
```
Comment (tranmanhlinh) [inline Assignment/Book-Management/src/app/core/services/book.service.ts:26]: - Store https://nga-book-api.herokuapp.com/api in environment.ts.
- Use template literals. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: Assignment/Book-Management/src/app/core/services/book.service.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      headers: new HttpHeaders({
+        Authorization: 'Bearer eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJuYW1laWQiOiIxIiwidW5pcXVlX25hbWUiOiJtYW5hZ2VyMSIsInJvbGUiOiJNQU5BR0VSIiwibmJmIjoxNTk1MzIwODk0LCJleHAiOjE1OTU3NTI4OTQsImlhdCI6MTU5NTMyMDg5NH0.PxPBRbpn75Jd8XVivxStWYiqUK6lT4-YA0o6HjcLfPAM_y0okSZxy3s3GDqlwxY6VuqvnBLtl5o3EXZ1AMUNdQ'
+      }),
+      observe: 'response'
+    }).pipe(
+      tap(resp => {
+        let objTemp = JSON.parse(resp.headers.get('pagination'));
+        this.totalPages = objTemp.totalPages;
+        this.totalItems = objTemp.totalItems;
+        console.log(this.totalItems);
```
Comment (tranmanhlinh) [inline Assignment/Book-Management/src/app/core/services/book.service.ts:36]: remove console.log

File: Assignment/Book-Management/src/app/core/services/book.service.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+
+      })).subscribe()
+
+    return this.http.get<any>('https://nga-book-api.herokuapp.com/api/books?pageNumber='+pageIndex+'&pageSize='+pageSize, {
+      headers: new HttpHeaders({
+        Authorization: 'Bearer eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJuYW1laWQiOiIxIiwidW5pcXVlX25hbWUiOiJtYW5hZ2VyMSIsInJvbGUiOiJNQU5BR0VSIiwibmJmIjoxNTk1MzIwODk0LCJleHAiOjE1OTU3NTI4OTQsImlhdCI6MTU5NTMyMDg5NH0.PxPBRbpn75Jd8XVivxStWYiqUK6lT4-YA0o6HjcLfPAM_y0okSZxy3s3GDqlwxY6VuqvnBLtl5o3EXZ1AMUNdQ'
+      })
```
Comment (tranmanhlinh) [inline Assignment/Book-Management/src/app/core/services/book.service.ts:44]: same.

File: Assignment/Book-Management/src/app/modules/book/pages/book-list/book-list.component.ts [hunk 5]
```diff
@@ -1,15 +1,61 @@
...
+  constructor(private bookService: BookService) {};
 
-  constructor() { }
+  displayedColumns: string[] = ['position', 'name', 'author', 'categories', 'cover', 'price', 'publisher', 'year'];
 
-  ngOnInit(): void {
+
+  ngOnInit() {
+    // this.dataSource.paginator = this.paginator;
+    this.bookObs = this.bookService.getAllBook(1,5);
```
Comment (tranmanhlinh) [inline Assignment/Book-Management/src/app/modules/book/pages/book-list/book-list.component.ts:42]: Use suffix $ for an observable name.

File: Assignment/Book-Management/src/app/modules/book/pages/book-list/book-list.component.ts [hunk 6]
```diff
@@ -1,15 +1,61 @@
...
   }
 
+  pageChange(event: PageEvent) {
+
+    this.bookObs = this.bookService.getAllBook(event.pageIndex+1, event.pageSize);
+    this.bookObs.subscribe(
+      val => {
+        this.dataSource = val;
+      }
+    )
```
Comment (tranmanhlinh) [inline Assignment/Book-Management/src/app/modules/book/pages/book-list/book-list.component.ts:59]: Use `async` pipe instead subscribe.

File: Assignment/Book-Management/src/app/modules/user/pages/login/login.component.ts [hunk 7]
```diff
@@ -1,15 +1,50 @@
...
+  {position: 2, name: 'Helium', weight: 4.0026, symbol: 'He'},
+  {position: 3, name: 'Lithium', weight: 6.941, symbol: 'Li'},
+  {position: 4, name: 'Beryllium', weight: 9.0122, symbol: 'Be'},
+  {position: 5, name: 'Boron', weight: 10.811, symbol: 'B'},
+  {position: 6, name: 'Carbon', weight: 12.0107, symbol: 'C'},
+  {position: 7, name: 'Nitrogen', weight: 14.0067, symbol: 'N'},
+  {position: 8, name: 'Oxygen', weight: 15.9994, symbol: 'O'},
+  {position: 9, name: 'Fluorine', weight: 18.9984, symbol: 'F'},
+  {position: 10, name: 'Neon', weight: 20.1797, symbol: 'Ne'},
+];
```
Comment (tranmanhlinh) [inline Assignment/Book-Management/src/app/modules/user/pages/login/login.component.ts:25]: What is it?

Comment (coonik) [reply Assignment/Book-Management/src/app/modules/user/pages/login/login.component.ts:25]: this is demo, 
i forgot to delete

File: Assignment/Book-Management/src/app/app-routing.module.ts [hunk 8]
```diff
@@ -1,8 +1,14 @@
...
 import { Routes, RouterModule } from '@angular/router';
+import { LoginComponent } from './modules/user/pages/login/login.component';
 
 
-const routes: Routes = [];
+const routes: Routes = [
+  {path: "", redirectTo: "book-list", pathMatch: "full"},
+  {path: "book-list", component: BookListComponent},
+  {path: "auth", component: LoginComponent}
+];
```
[the remark this answers is not in the document]
Comment (coonik) [reply Assignment/Book-Management/src/app/app-routing.module.ts:14]: done

File: Assignment/Book-Management/src/app/core/services/book.service.ts (hunk 1 above)
Comment (coonik) [reply Assignment/Book-Management/src/app/core/services/book.service.ts]: done

File: Assignment/Book-Management/src/app/core/services/book.service.ts (hunk 2 above)
Comment (coonik) [reply Assignment/Book-Management/src/app/core/services/book.service.ts]: done

File: Assignment/Book-Management/src/app/core/services/book.service.ts (hunk 3 above)
Comment (coonik) [reply Assignment/Book-Management/src/app/core/services/book.service.ts]: done

File: Assignment/Book-Management/src/app/core/services/book.service.ts (hunk 4 above)
Comment (coonik) [reply Assignment/Book-Management/src/app/core/services/book.service.ts:33]: done

File: Assignment/Book-Management/src/app/modules/book/pages/book-list/book-list.component.ts (hunk 5 above)
Comment (coonik) [reply Assignment/Book-Management/src/app/modules/book/pages/book-list/book-list.component.ts:47]: done

File: Assignment/Book-Management/src/app/modules/book/pages/book-list/book-list.component.ts (hunk 6 above)
Comment (coonik) [reply Assignment/Book-Management/src/app/modules/book/pages/book-list/book-list.component.ts:64]: Done

**Pull request merged**